core/states/approaching_marker.py

```diff
@@ -14,13 +14,10 @@ class ApproachingMarker(RoverState):
 
     def start(self):
         # Schedule AR Tag detection
-        loop = asyncio.get_event_loop()
-        self.ar_tag_task = loop.create_task(core.vision.ar_tag_detector.async_ar_tag_detector())
         self.num_detection_attempts = 0
 
     def exit(self):
         # Cancel all state specific coroutines
-        self.ar_tag_task.cancel()
```

Comment:
This may be more involved than it seems, but why is it no longer possible to exit ar tag detection? My assumption is because we want to stream it even if we don't want to actually handle detection, but I'd enjoy the clear answer and how it affects autonomy performance.

Reply:
Yeah, so I initially planned to have the coroutines started/exited for each state that needed them so we could technically reduce the performance impact of running AR Tag/Obstacle Avoidance together at all times. However this makes observing the results in testing very hard as we are constantly interrupting our streams. This is something we could easily change in the future if we need better performance for a competition sim, but for right now the better observability wins.